<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="bag_file" default="/home/chenxifeng/Documents/sensor_calibration/data_rr2/rr2newright.bag"/>
  <arg name="transforms_from_csv" default="false"/>
  <arg name="csv_file" default="PATH/TO/YOUR.csv"/>
  <arg name="mode" default="Point2Plane"/>  

  <node pkg="lidar_align" name="lidar_align" type="lidar_align_node" output="screen">
    <param name="input_bag_path" value="$(arg bag_file)" />
    <param name="input_csv_path" value="$(arg csv_file)" />
    <param name="local" value="true" />
    <!-- <param name="transforms_from_csv" value="$(arg transforms_from_csv)"/> -->
    <!-- rr1 left lidar to imu -->
    <!-- <rosparam param="inital_guess">[1.73, 1.3125, -0.67, -0.02, -0.015, -0.02, 0.0]</rosparam> -->
    <!-- rr1 right lidar to imu -->
    <!-- <rosparam param="inital_guess">[1.73, -1.4125, -0.66, -0.01, -0.02, -0.0545, 0.0]</rosparam>; -->
    <!-- rr2 left lidar to imu -->
    <!-- <rosparam param="inital_guess">[1.7, 1.3, -0.7, 0.01, -0.02, -0.01, 0.0]</rosparam> -->
    <!-- rr2 right lidar to imu -->
    <rosparam param="inital_guess"> [1.7714, -1.28892, -0.757679, 0.0201012, -0.0155248, -0.0204949, 0.0]</rosparam>
    <param name="max_time_offset" value="0.1" />
    <param name="angular_range" value="0.05" />
    <param name="translation_range" value="0.2" />
    <param name="max_evals" value="200" />
    <param name="xtol" value="0.0001" />
    <param name="knn_batch_size" value="1000" />
    <param name="knn_k" value="5" />
    <param name="global_knn_max_dist" value="1.0" />
    <param name="local_knn_max_dist" value="0.5" />
    <param name="time_cal" value="true" />
    <param name="mode" value="$(arg mode)" />
    <param name="output_pointcloud_path" value="$(find lidar_align)/results/$(anon lidar_points).ply" />
    <param name="output_calibration_path" value="$(find lidar_align)/results/$(anon calibration).txt" />
    <param name="leaf_size" value="0.1" />
    <param name="use_n_scans" value="2000" />
  </node>
  
</launch>